src/test/ui/nll/return-ref-mut-issue-46557.stderr

```diff
@@ -1,11 +1,16 @@
 error[E0597]: borrowed value does not live long enough
   --> $DIR/return-ref-mut-issue-46557.rs:17:21
    |
-LL |     let ref mut x = 1234543; //~ ERROR borrowed value does not live long enough [E0597]
-   |                     ^^^^^^^ temporary value does not live long enough
-LL |     x
-LL | }
-   | - temporary value only lives until here
+LL |   fn gimme_static_mut() -> &'static mut u32 {
+   |  ___________________________________________-
+LL | |     let ref mut x = 1234543; //~ ERROR borrowed value does not live long enough [E0597]
+   | |                     ^^^^^^^ temporary value does not live long enough
+LL | |     x
+LL | | }
+   | | -
+   | | |
+   | |_temporary value only lives until here
+   |   borrow later used here
```

Comment:
This is very odd looking.

It seems to me that 1) there's a bug in the graphing code for a multiline span ending in the same char as a single line span, and 2) we should probably only show one of these spans, as they do not add more information.

Personally, I would prefer the output to look more like this:

```
error[E0597]: borrowed value does not live long enough
  --> $DIR/return-ref-mut-issue-46557.rs:17:21
   |
LL | fn gimme_static_mut() -> &'static mut u32 {
   |                          ---------------- the returned value needs to live for the duration of the `'static` lifetime
LL |     let ref mut x = 1234543;
   |                     ^^^^^^^ value does not live long enough
LL |     x
   |     - value is being returned here
LL | }
   | - value only lives until here
```

Reply:
@estebank @nikomatsakis I thought exactly the same thing. Waht about opening a different issue to avoid tackling a lot of different things here?.

Reply:
👍